### ISSUE TITLE
使用arcfaceloss无法收敛

### ISSUE THREAD

**Issue by dl8207531** [auroua/InsightFace_TF#42]
我将arcfaceloss函数用到我的代码中，mobilefacenet网络无法收敛，其他网络loss为nan，但是使用softmax没有问题，请问这里有什么需要注意的吗，谢谢？

**Comment by ltcs11** [comment]
可以先使用softmax进行预训练，之后使用arcface进行fine tune
在原作者的issue中有过相关讨论
[https://github.com/deepinsight/insightface/issues/214](url)

**Comment by Joll123** [comment]
@ltcs11 为什么使用arcface训练时一直显示Accuracy为0，
![image](https://user-images.githubusercontent.com/43200569/47843050-6a9bf100-ddf9-11e8-8ae6-2af8a406d3ec.png)
谢谢

**Comment by ltcs11** [comment]
@Joll123 
你是直接使用softmax预训练之后再接arcface fine tune吗？
如果是，可以在继续训练一段时间，或增大batch_size

**Comment by dl8207531** [comment]
不用softmax预训练应该也能收敛，只是效果不好。我那个不收敛是因为minibatch太小只有128，学习率又太大导致。减小学习率就能收敛了

**Comment by Joll123** [comment]
@dl8207531 你取的学习率多大

**Comment by jetsmith** [comment]
@dl8207531 老兄用哪个数据集训练的, margin多少?训练的模型在lfw的准确率能达到多少呢?

**Comment by dl8207531** [comment]
@Joll123 用softmax预训练可以从0.01开始 如果想测试直接mobilefacenet训练可以试下也从0.01开始，如果不行就0.001肯定能收敛

**Comment by dl8207531** [comment]
@jetsmith 就InsightFace提供的那个数据集 margin 0.5 mobilefacenet 准确率99.367% 达不到作者baseline 我猜原因应该是minibatch太小

**Comment by jetsmith** [comment]
@dl8207531  那你用softmax预训练训到什么程度转arcface训练呢, 你arcface学习率的schedule是如何设置的呢?

**Comment by dl8207531** [comment]
@jetsmith softmax从0.01开始训练到不再收敛保持该学习率改用arc，arc同样训练到不再收敛学习率乘0.1继续训练

**Comment by zephyr737** [comment]
@dl8207531 您好，请问 你后面改用arc时是全部参数训练，还是冻结了卷积层呢？

**Comment by BarryLYH** [comment]
Based on my training, it is hard to converge loss to a small number if train with m=0.5. My solution is that train with m =0 at the beginning and then increase m to 0.1, 0.2, 0.4, 0.5. It works well and the loss goes from 6 to 0.3 in my database.

**Comment by muyuuuu** [comment]
> and then increase m to 0.1, 0.2, 0.4, 0.5.

How about the frequent of increasing m? 

- Epoch [0-10] m is 0, [10-20] m is 0.1, ... epoch [40-50] m is 0.5. If I do this, my losses will increase after the adjustment.
- or epoch 1 m is 0, epoch 2 m is 0.01, ..., epoch 50 m is 0.5 ?

And lr gradually decay or remain constant at a small value?

**Comment by longtaojiang** [comment]
很有用

**Comment by bigbigbigwaterbucket** [comment]
> Based on my training, it is hard to converge loss to a small number if train with m=0.5. My solution is that train with m =0 at the beginning and then increase m to 0.1, 0.2, 0.4, 0.5. It works well and the loss goes from 6 to 0.3 in my database.

Why doe this solution work?
